02-microarray/pathway-analysis_microarray_02_ora_with_webgestaltr.Rmd

```diff
@@ -0,0 +1,262 @@
+The table contains Ensembl gene IDs, log fold-changes for each group, and adjusted p-values (FDR in this case).
+We can identify upregulated genes by filtering these results and use this list as input to ORA.
+
+```{r}
+# Define the url to your differential expression results file
+dge_url <- "https://refinebio-examples.s3.us-east-2.amazonaws.com/02-microarray/results/GSE37418/GSE37418_limma_results.tsv"
+# Replace with a link to your desired differential expression results TSV file
+```
+
+Read in the online file that has differential expression results. 
+
+```{r}
+# Read in the contents of your differential expression results file
+dge_df <- readr::read_tsv(dge_url) # `dge_url` can be replaced with a file path to a TSV file with your desired gene list results
+```
+
+`read_tsv()` can read TSV files online and doesn't necessarily require you download the file first. 
+However, should you decide you want to download the file to your `data/` directory, you can use `download.file(dge_url, destfile = file.path(data_dir, "GSE37418_limma_results.tsv"))`.
+
+Let's take a look at what these differential expression results look like. 
+
+```{r}
+dge_df
+```
+
+Here we'll use the `SHH` groups' log fold-change > 2 and FDR < 0.05 as cutoffs for determining 
+what genes are of interest.
+
+```{r}
+upregulated_genes <- dge_df %>%
+  dplyr::filter(SHHvsOther > 2, adj.P.Val < 0.05) %>%
+  dplyr::pull(Gene)
+```
+
+Because we are testing if there is more overlap between a set of genes of interest and gene sets or pathways from some knowledge base (e.g., KEGG, Gene Ontology (GO)) than we would expect at random, we need to identify an appropriate background set. 
+Put another way, if a gene is _not measured_, it can not possibly be in our gene set of interest.
+We can provide our analysis method of choice with a reference list, which we will generate by selecting all the genes in the table of differential expression results.
+
+```{r}
+all_genes <- dge_df %>%
+  dplyr::pull(Gene)
+```
+
+## WebGestaltR
+
+We can check whether or not we need to convert to a different gene identifier by figuring out what gene identifiers `WebGestaltR` accepts for human.
+We can do this with the `listIdType()` function; the first argument is the organism name.
+
+```{r}
+listIdType("hsapiens")
+```
+
+We can see that there is `ensembl_gene_id` listed above so the Ensembl IDs in the `Gene` column are compatible identifiers, and we do not need to convert to a different identifier.
+All refine.bio dataset downloads come with Ensembl IDs but if do find yourself needing to convert gene IDs, we'd encourage you to look at our [gene ID annotation examples](https://alexslemonade.github.io/refinebio-examples/02-microarray/gene-id-annotation_microarray_01_ensembl.html).
+
+The `WebGestaltR` function is a wrapper for the [WEB-based GEne SeT AnaLysis Toolkit (WebGestalt)](http://www.webgestalt.org/); note that WebGestalt has a new 2019 version) [@Wang2017].
```

Comment:
Your `note that WebGestalt has a new 2019 version` means that this is a newer version than what is being used in the wrapper function? If so can we clarify this here?

Reply:
This is something I brought over from Jackie's original notebook, I'm not really sure it still applies or that it is really info users need to know, so I'm going to delete it. Unless she later says its still a needed note.